File: MyNote2/MyNote2/MasterViewController.m
//
//  MasterViewController.m
//  MyNote2
//
//  Created by 俊杰李 on 2020/5/13.
//  Copyright © 2020 俊杰李. All rights reserved.
//

#import "MasterViewController.h"

@interface MasterViewController ()
//保存数据列表
@property (nonatomic,strong) NSMutableArray *allList;
@end

@implementation MasterViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.navigationItem.leftBarButtonItem = self.editButtonItem;
    
    self.detailViewController = (DetailViewController *)[[self.splitViewController.viewControllers lastObject] topViewController];
    [ self loadDate];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(reloadDate:) name:@"reload" object:nil];
}

#pragma mark - 数据处理
- (void)loadDate{
    NoteDao *noteDao = [NoteDao sharedInstance];
    self.allList =[noteDao findAll];
}

- (void)reloadDate:(NSNotification*)notification{
    NSMutableArray *resList = [notification object];
    self.allList  = resList;
    [self.tableView reloadData];
}

#pragma mark - TableView处理

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    
    Note* note = self.allList[indexPath.row];
    cell.textLabel.text = note.content;
    cell.detailTextLabel.text = [note.date description];
    //    cell.textLabel.text = @"note.content";
    //    cell.detailTextLabel.text = @"note.date";
    
    return cell;
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.allList.count;
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath{
    return YES;
}

- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath{
    if(editingStyle == UITableViewCellEditingStyleDelete){
        Note* note = self.allList[indexPath.row];
        NoteDao* dao = [NoteDao sharedInstance];
        [dao remove:note];
        self.allList = [dao findAll];
        
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    }
}


- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"showDetail"]) {
        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
        Note *note = self.allList[indexPath.row];
        DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
        [controller setDetailItem:note];
        controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
        controller.navigationItem.leftItemsSupplementBackButton = YES;
    }
}


/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

@end

File: MyNotes-swift-Alamofire/MyNotes-swift-Alamofire/MasterViewController.swift
//
//  MasterViewController.swift
//  MyNotes-swift-Alamofire
//
//  Created by 俊杰李 on 2020/5/19.
//  Copyright © 2020 俊杰李. All rights reserved.
//

import UIKit
import Alamofire
class MasterViewController: UITableViewController {
    
    var detailViewController: DetailViewController? = nil
    var notes:NSArray = NSArray()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        navigationItem.leftBarButtonItem = editButtonItem
        
        if let split = splitViewController {
            let controllers = split.viewControllers
            detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
        }
        let rc = UIRefreshControl()
        rc.attributedTitle = NSAttributedString(string: "下拉刷新");
        rc.addTarget(self, action: #selector(loadNotes), for: UIControl.Event.valueChanged)
        self.refreshControl = rc;
        
        NotificationCenter.default.addObserver(self, selector: #selector(loadNotes), name: NSNotification.Name("reload"), object: nil)
        loadNotes();
    }
    
    override func viewWillAppear(_ animated: Bool) {
        clearsSelectionOnViewWillAppear = splitViewController!.isCollapsed
        super.viewWillAppear(true)
        self.loadNotes()
    }
    
    // MARK: - Segues
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "showDetail" {
            if let indexPath = tableView.indexPathForSelectedRow {
                let note = notes[indexPath.row] as! NSDictionary
                let controller = (segue.destination as! UINavigationController).topViewController as! DetailViewController
                controller.detailItem = note
                controller.navigationItem.leftBarButtonItem = splitViewController?.displayModeButtonItem
                controller.navigationItem.leftItemsSupplementBackButton = true
                detailViewController = controller
            }
        }
    }
    
    // MARK: - Table View
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return notes.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        let object = notes[indexPath.row] as! NSDictionary
        let content = object["Content"] as! String
        let date = object["CDate"] as! String
        
        cell.textLabel!.text = content
        cell.detailTextLabel!.text = date
        return cell
    }
    
    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        // Return false if you do not want the specified item to be editable.
        return true
    }
    
    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
        if editingStyle == .delete {
            let object = notes[indexPath.row] as! NSDictionary
            //            tableView.deleteRows(at: [indexPath], with: .fade)
            deleteNote(object)
        } else if editingStyle == .insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view.
        }
    }
    
    func deleteNote(_ note:NSDictionary){
        let strURL = "http://www.51work6.com/service/mynotes/WebService.php"
        let params:Parameters = ["email": "l536510961@126.com", "type": "JSON", "action": "remove","id":note["ID"] ?? "0"]
        AF.request(strURL,method: .post,parameters: params    )
            .validate(statusCode: 200..<300)
            .validate(contentType: ["application/json"])
            .responseJSON { response in
                switch response.result {
                case .success:
                    self.loadNotes()
                    break
                case let .failure(error):
                    print(error)
                    break
                }
                
        }
    }
    
    @objc func loadNotes(){
        if(self.refreshControl!.isRefreshing){
             self.refreshControl!.attributedTitle = NSAttributedString(string: "加载数据中...")
        }
        UIApplication.shared.isNetworkActivityIndicatorVisible  = true
        print("查询列表.....")
        let strURL = "http://www.51work6.com/service/mynotes/WebService.php"
        let params:Parameters  = ["email": "l536510961@126.com", "type": "JSON", "action": "query"]
        
        AF.request(strURL,method: .post,parameters: params)
            .validate(statusCode: 200..<300)
            .validate(contentType: ["application/json"])
            .responseJSON { response in
                switch response.result {
                case .success:
                    let data =  response.value as! NSDictionary
                    self.notes = data["Record"] as! NSArray
                    self.tableView.reloadData()
                case let .failure(error):
                    print(error)
                }
                self.refreshControl!.endRefreshing()
                self.refreshControl!.attributedTitle = NSAttributedString(string: "下拉刷新")
                 UIApplication.shared.isNetworkActivityIndicatorVisible  = false
        }
    }
}

